### DISCUSSION TITLE
Reset a MUI Switch

### DISCUSSION THREAD

**Discussion by niicojs** [react-hook-form/react-hook-form#2820] [Q&A] [answered]
Hi, 
I'm trying to reset an uncontrolled Material UI Switch to it's default state (unchecked) but can't do.
Reset works for mui textfield but not switch.
 
I've tried several way to register it. 
Current version is:
```
            <FormControlLabel
              control={<Switch name="viewdetails" value={1} ref={register} />}
              label="View Details"
            />```

Thanks

**Comment by bluebill1049** [comment, marked as answer]
Take a look at this CSB: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r you should use Controller.

**Reply by niicojs** [reply]
That was fast, thanks a lot!
This example is great.

What I can't see is how to combined `value` and `checked` for switch (if checked it sends value, not if unchecked).
But I can definitly live without that :)

Here is what I did for the record:
```
            <Controller
              name="viewdetails"
              control={control}
              render={({ value, onChange }) => (
                <FormControlLabel
                  control={
                    <Switch
                      onChange={(e) => onChange(e.target.checked)}
                      checked={value}
                    />
                  }
                  label="View details"
                />
              )}
            />
 ```
